### ISSUE TITLE
[Bug]: Failed to index workspace folder(s)

### ISSUE THREAD

**Issue by pulupa** [interactive-data-language/vscode-idl#15]
### Current Behavior

When starting the IDL for VSCode extension, the server does not index the folders in the workspace.

### Expected Behavior

The extension should index files and print information in the log.

### Steps to Reproduce

1. In a new VSCode window, create an IDL file. This starts the extension. Since no folders have been added to the workspace, the only folder to index is .vscode/extensions/idl.idl-for-vscode-4.0.1/idl/vscode/notebooks. The code reports a failure to index.

### Failure Logs

```shell
idl-lsp info Server initialized, indexing code in these folders:
  {
    '/Users/pulupa/.vscode/extensions/idl.idl-for-vscode-4.0.1/idl/vscode/notebooks': true
  }
idl-lsp info Attempting to index 22 PRO file(s)
idl-lsp error Failed to index workspace folder(s)
  {
    message: [ 'Invalid array length' ],
    stack: [
      'RangeError: Invalid array length',
      '    at t.SplitFiles (/Users/pulupa/.vscode/extensions/idl.idl-for-vscode-4.0.1/dist/apps/server/main.js:7:196089)',
      '    at IDLIndex.indexWorkspaceProFiles (/Users/pulupa/.vscode/extensions/idl.idl-for-vscode-4.0.1/dist/apps/server/main.js:1:521851)',
      '    at IDLIndex.indexWorkspace (/Users/pulupa/.vscode/extensions/idl.idl-for-vscode-4.0.1/dist/apps/server/main.js:1:522837)',
      '    at async /Users/pulupa/.vscode/extensions/idl.idl-for-vscode-4.0.1/dist/apps/server/main.js:7:261223'
    ]
  }
```

### Operating System

- [ ] macOS (Intel x86_64)
- [X] macOS (ARM64)
- [ ] Linux
- [ ] Windows

### IDL Version

8.9

### Additional Information

For me, this error is occurring on a new install of VSCode on an M3 MacBook Pro, and did not occur on Intel Macs.

I get similar behavior whether I open a workspace which has folders, or the minimal example above, where the only files to index are in the .vscode/extensions/idl.idl-for-vscode-4.0.1/idl/vscode/notebooks folder.

The rest of the extension (debug, compile, run, hover help for built-in routines) is working normally.

I do have node installed (tried with version 18 and version 21)

**Comment by znorman-harris** [comment]
After looking at this, I think I have a fix for the problem which I think boils down to a rounding error (lack of rounding) when getting CPU count. Until the latest generation of Macs came out, I think we have always had even CPU counts, but there is an M3 Mac with 11 CPUs.

To confirm this is the problem, does your Mac have an odd number of CPUs?

**Comment by pulupa** [comment]
Thanks for the quick response! Yes, the Mac is an M3 Pro with 11 cores.

**Comment by znorman-harris** [comment]
Alright, just published a new release (4.1.0) which should properly handle odd-numbered CPU core counts.

Let us know if you have any other problems as we are happy to take a look!
